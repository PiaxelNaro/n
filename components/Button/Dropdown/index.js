// @flow
import React from 'react';
import styled from 'styled-components';
import DropdownIcon from 'react-icons/lib/md/arrow-drop-down';
import Text from '../../Text';
import FlexGrid from '../../FlexGrid';
import Button from '..';

const StyledDropdownButton = styled(Button)`
    text-align: left;
    width: 100%;
`;

const DropdownLabel = styled(FlexGrid)`
    color: #999;
    font-size: 11px;
    align-items: flex-start;
`;

const DropdownButton = ({
    label,
    labelIcon,
    value,
    width = '110px',
}: {
    label?: string,
    labelIcon?: any,
    value: string,
    width?: string,
}) => {
    let buttonLabel;
    if (label) {
        buttonLabel = (
            <DropdownLabel gutter="3px">
                {labelIcon}
                <Text ellipsis>{label}</Text>
            </DropdownLabel>
        );
    }

    return (
        <div style={{ width }}>
            {buttonLabel}
            <StyledDropdownButton>
                <FlexGrid>
                    <FlexGrid.Item>
                        <Text ellipsis>{value}</Text>
                    </FlexGrid.Item>
                    <DropdownIcon size={13} />
                </FlexGrid>
            </StyledDropdownButton>
        </div>
    );
};

export default DropdownButton;
