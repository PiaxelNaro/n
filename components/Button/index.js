// @flow
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';

const StyledButton = styled.button.attrs({
    className: props =>
        classnames({
            'is-active': props.active,
            'is-disabled': props.disabled,
        }),
    disabled: props => props.disabled,
})`
    cursor: pointer;
    background: #efefef;
    background-image: linear-gradient(to top, #efefef, #fff);
    border: 1px solid #ccc;
    border-radius: 2px;
    padding: 8px 7px;
    outline: none;
    text-decoration: none;
    color: #000;
    font-size: 11px;
    font-family: Helvetica, Arial, sans-serif;

    &.is-active {
        background-image: linear-gradient(to bottom, #efefef, #fff);
    }

    &.is-disabled {
        color: #a5a5a5;
        cursor: not-allowed;
    }
`;

const StyledLink = StyledButton.withComponent('a');

const Button = (props: { href?: string }) => {
    if (props.href) {
        return <StyledLink {...props} />;
    }
    return <StyledButton {...props} />;
};

Button.Blank = styled.button`
    border: none;
    background: none;
    font-size: 13px;
    padding: 0;
    margin: 0;
    cursor: pointer;
    outline: none;
    color: inherit;
`;

Button.Action = styled.button`
    border: 1px solid #3F88D4;
    border-radius: 15px;
    font-size: 13px;
    min-width: 60px;
    height: 27px;
    padding: 0 20px;
    cursor: pointer;
    outline: none;
    background: none;
`;

Button.Primary = styled(Button.Action)`
    background-color: #3F88D4;
    color: #FFF;
    cursor: pointer;
    line-height: 1;

    :hover,
    :focus {
        background-color: #65A0DD;
    }

    &[disabled] {
        background-color: #3F88D4;
        opacity: 0.5;
        cursor: default;
    }
`;
Button.Primary.displayName = 'ButtonPrimary';

Button.Secondary = styled(Button.Action)`
    color: #3F88D4;
    background-color: #FFF;
    cursor: pointer;
    line-height: 1;

    :hover,
    :focus {
        color: #FFF;
        background-color: #3F88D4;
    }

    &[disabled] {
        color: #3F88D4;
        background-color: #FFF;
        opacity: 0.5;
        cursor: default;
    }
`;
Button.Secondary.displayName = 'ButtonSecondary';

export default Button;
export { StyledButton, StyledLink };
