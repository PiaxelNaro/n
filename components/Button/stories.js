// @flow
import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import ClearIcon from 'react-icons/lib/md/clear';
import GlobeIcon from 'react-icons/lib/fa/globe';
import Table from '../Table';
import FlexGrid from '../FlexGrid';
import DropdownButton from './Dropdown';
import Button from '.';

storiesOf('Button', module)
    .add('button types', () =>
        <Table
            fixedColumns={0}
            columns={[
                {
                    label: 'Chrome button',
                    key: 'chrome',
                },
                {
                    label: 'Action Button',
                    key: 'action',
                },
                {
                    label: 'Dropdown Button',
                    key: 'dropdown',
                },
            ]}
            items={[
                {
                    id: 'example',
                    chrome: (
                        <div>
                            <FlexGrid gutter="4px">
                                <Button>Default</Button>
                                <Button active>Active</Button>
                                <Button disabled>Disabled</Button>
                            </FlexGrid>
                            <FlexGrid gutter="4px">
                                <Button href="#">Default</Button>
                                <Button href="#" active>Active</Button>
                                <Button href="#" disabled>Disabled</Button>
                            </FlexGrid>
                        </div>
                    ),
                    action: (
                        <FlexGrid gutter="4px">
                            <Button.Primary>Primary</Button.Primary>
                            <Button.Secondary>Secondary</Button.Secondary>
                            <Button.Primary disabled>Disabled</Button.Primary>
                            <Button.Secondary disabled>Disabled</Button.Secondary>
                        </FlexGrid>
                    ),
                    dropdown: (
                        <FlexGrid gutter="4px">
                            <DropdownButton
                                label="Country"
                                labelIcon={<GlobeIcon size={11} />}
                                value="China"
                            />
                        </FlexGrid>
                    ),
                },
            ]}
        />
    )
    .add('blank button with icon', () =>
        <div>
            <Button.Blank onClick={action('Clicked icon button')}>
                <ClearIcon size={16} />
            </Button.Blank>
        </div>
    );
