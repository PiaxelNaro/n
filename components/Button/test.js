// @flow
import React from 'react';
import { shallow } from 'enzyme';
import ClearIcon from 'react-icons/lib/md/clear';

import Button, { StyledButton, StyledLink } from '.';

describe('<Button>', () => {
    it('renders text', () => {
        expect(shallow(<Button>My Button</Button>)).toMatchSnapshot();
        expect(shallow(<Button active>My Button</Button>)).toMatchSnapshot();
        expect(shallow(<Button disabled>My Button</Button>)).toMatchSnapshot();
        expect(shallow(<Button href="testLink">My Button</Button>)).toMatchSnapshot();
    });
});

describe('<StyledButton>', () => {
    expect(
        shallow(<StyledButton active>My Button</StyledButton>)
    ).toMatchSnapshot();
    expect(
        shallow(<StyledButton disabled>My Button</StyledButton>)
    ).toMatchSnapshot();
});

describe('<StyledLink>', () => {
    expect(
        shallow(<StyledLink active href="testLink">My Button</StyledLink>)
    ).toMatchSnapshot();
    expect(
        shallow(<StyledLink disabled href="testLink">My Button</StyledLink>)
    ).toMatchSnapshot();
});

describe('<Button.Blank>', () => {
    const clickSpy = jest.fn();
    const button = shallow(<Button.Blank onClick={clickSpy}><ClearIcon /></Button.Blank>);

    it('renders button with icon', () => {
        expect(button).toMatchSnapshot();
    });

    it('click blank button', () => {
        button.simulate('click');
        expect(clickSpy).toHaveBeenCalled();
    });
});
