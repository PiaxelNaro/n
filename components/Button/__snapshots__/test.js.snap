// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className="Button__StyledButton-s1auny4n-0 is-active bFaiIv"
>
  My Button
</button>
`;

exports[` 2`] = `
<button
  className="Button__StyledButton-s1auny4n-0 is-disabled bFaiIv"
  disabled={true}
>
  My Button
</button>
`;

exports[` 3`] = `
<a
  className="sc-bdVaJa is-active laxZm"
  href="testLink"
>
  My Button
</a>
`;

exports[` 4`] = `
<a
  className="sc-bdVaJa is-disabled laxZm"
  disabled={true}
  href="testLink"
>
  My Button
</a>
`;

exports[`<Button.Blank> renders button with icon 1`] = `
<button
  className="Button__Blank-s1auny4n-1 gBWxu"
  onClick={[Function]}
>
  <MdClear />
</button>
`;

exports[`<Button> renders text 1`] = `
<Button__StyledButton>
  My Button
</Button__StyledButton>
`;

exports[`<Button> renders text 2`] = `
<Button__StyledButton
  active={true}
>
  My Button
</Button__StyledButton>
`;

exports[`<Button> renders text 3`] = `
<Button__StyledButton
  disabled={true}
>
  My Button
</Button__StyledButton>
`;

exports[`<Button> renders text 4`] = `
<styled.a
  href="testLink"
>
  My Button
</styled.a>
`;
